pkg/services/secrets/manager/manager.go

```diff
@@ -358,16 +360,30 @@ var (
 
 func (s *SecretsService) Run(ctx context.Context) error {
 	gc := time.NewTicker(gcInterval)
+	grp, gCtx := errgroup.WithContext(ctx)
+
+	for _, p := range s.providers {
+		if svc, ok := p.(registry.BackgroundService); ok {
```

Comment:
I'd suggest adding a new interface that duplicated `registry.BackgroundService`, reusing the interface signals that this code is somehow related with wire and the registry when it actually isn't.

Reply:
Thank you. Done :)